MyBus/MyBus/MainViewController.swift

```diff
@@ -96,8 +96,18 @@ class MainViewController: UIViewController, MapBusRoadDelegate {
             self.cycleViewController(self.currentViewController!, toViewController: self.mapViewController)
             self.currentViewController = self.mapViewController
         }
-        self.mapViewController.mapView.showsUserLocation = true
-        self.mapViewController.mapView.setZoomLevel(16, animated: false)
+        //check the status of localization        
+        if (self.mapViewController.mapView.userLocation != nil)
```

Comment:
@julietapoume seems better. To consider case of location services disabled for all apps I suggest replace with this code.

	let locationServiceAuth = CLLocationManager.authorizationStatus()
    guard (locationServiceAuth == .AuthorizedAlways || locationServiceAuth == .AuthorizedWhenInUse) else {
        let alert = UIAlertController.init(title: "No te encontramos en el mapa", message: "Si queres que te ubiquemos en el mapa activa tu localizacion.", preferredStyle: .Alert)
        let action = UIAlertAction.init(title: "OK", style: UIAlertActionStyle.Default, handler: nil)
        alert.addAction(action)
        return self.presentViewController(alert, animated: true, completion: nil)
    }
    
    self.mapViewController.mapView.showsUserLocation = true
    self.mapViewController.mapView.setZoomLevel(16, animated: false)